arch/cortex-m4/src/mpu.rs

```diff
@@ -163,8 +163,18 @@ impl kernel::mpu::MPU for MPU {
         } else {
             // Memory base not aligned to memory size
 
-            // Which subregion size would align with the base address?
-            let subregion_size = start % len;
+            // Which (power-of-two) subregion size would align with the base
+            // address?
+            //
+            // We find this by taking smallest binary substring of the base
+            // address with exactly one bit:
+            //
+            //      1 << (start.trailing_zeros())
+            //
+            // The zero case (where there are no 1 bits) is taken care of with
+            // the `wrapping_shl` operation, which masks bits that would
+            // overflow the integer width.
+            let subregion_size = (1 as usize).wrapping_shl(start.trailing_zeros());
```

Comment:
Yes, I don't believe we would ever realistically get into a case where the `wrapping_shl` matters. However, it seemed worthwhile to do this for "defense-in-depth" in case of future refactoring. In practice it's an additional 2 instructions vs a vanilla `<<`.